auth/acquire.go

```diff
@@ -0,0 +1,10 @@
+// SPDX-FileCopyrightText: 2025 Comcast Cable Communications Management, LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+// Acquirer acquires the credential for http request authorization headers.
+type Acquirer interface {
+	// Acquire gets a credential string.
+	Acquire() (string, error)
```

Comment:
I think you'll be happier with a slightly different interface.  Below is nice because it defers to the thing with the secret about how it gets added.  So if it's `Authentication: bearer XYZ` great.  If it's some custom header set, the code in this library doesn't care.

```golang
type Decorator interface {
        Decorate(ctx context.Context, req *http.Request) error
}
```

Reply:
that's such a great point!